tpch_gen.py

```diff
@@ -0,0 +1,93 @@
+import tempfile
+import psutil
+import pathlib
+import duckdb
+import botocore.session
+import coiled
+import click
+
+
+@click.command()
+@click.option("--sf", default=10, help="Scale factor to use")
+@click.option("--target-mb", default=50, help="Target output parquet file size")
+@click.option(
+    "--path",
+    default="./tpch-data",
+    help="Local or S3 base path, will affix 'scale-<scale>' subdirectory to this path",
+)
+@coiled.function(region="us-east-2", memory="512 GiB", disk_size=500)
+def main(sf, target_mb, path):
```

Comment:
My preference would be to have this also accessible as a function that people can call from Python.  I think that this will make this code more flexibly useful.  

Maybe one way to do this would be to pull out most of the body of the `main` code to a standalone function, and then to lightly wrap that function in a main function with all the click decorators.  That way we could do both. Thoughts?

Also, the names `sf`, `target_mb` are not immediately obvious to me.  I suggest instead `scale_factor` and `target_size` (but other names are probably good as long as they're understandable without having to refer to docs or comments).

Reply:
Ya it's rough.. feel free to push any changes you like when playing with it. I'll make some further updates tomorrow.